lib/commands/implementations/banphrase.js

```diff
@@ -38,6 +43,7 @@ function banphrase(defaultPunishmentDuration, punishmentType) {
 }
 
 module.exports = {
-  addban: new Command(banphrase(1800, 'ban'), true, true, /(\d+[HMDSWwhmds])?\s?(.*)/, false),
-  addmute: new Command(banphrase(600, 'mute'), true, true, /(\d+[HMDSWwhmds])?\s?(.*)/, false),
+  addban: new Command(banphrase('twitch', 1800, 'ban'), true, true, /(\d+[HMDSWwhmds])?\s?(.*)/, false),
+  addmute: new Command(banphrase('twitch', 600, 'mute'), true, true, /(\d+[HMDSWwhmds])?\s?(.*)/, false),
+  addphrase: new Command(banphrase('dgg'), false, true),
```

Comment:
I don't think it makes sense to add a new command when one of the goals of banned phrases is to integrate chat-bot features into the chat backend. For Dgg chat, addban and addmute are legacy interfaces that we only continue to maintain to ease the transition. With that in mind, I prefer an approach where addmute and addban still exist for Dgg chat, but just ignore the duration/punishment type and issue `ADDPHRASE` commands internally. You don't need addphrase.

Beyond that, it might be better to create two separate implementations of the phrase commands, one for Dgg and one for Twitch, and conditionally assign the right one during initialization. That reads a little cleaner than having a single implementation that conditionally changes its behavior depending on which chat it's connecting to.

Reply:
Sounds reasonable, I removed the `!*phrase` commands and refactored the code as per your suggestion. Just to double-check: the d.gg commands should be exactly the same as before (as in, the duration will be expected by the command, despite ultimately not being used)?

Reply:
Either is fine assuming the bot doesn't complain if extra arguments are passed in.